### PR TITLE
feat: model devi C/C++ API without nlist

### PR DESCRIPTION
**Pull request by robinzyb**
To solve this issue:
https://github.com/deepmodeling/deepmd-kit/issues/3641

### CODE REVIEW

File: source/api_cc/include/DeepPot.h [hunk 1]
```diff
@@ -629,6 +629,36 @@ class DeepPotModelDevi {
...
+   * nframes x natoms x dim_aparam.
+   * natoms x dim_aparam. Then all frames are assumed to be provided with the
+   *same aparam. dim_aparam. Then all frames and atoms are provided with the
+   *same aparam.
+   **/
+  template <typename VALUETYPE>
+  void compute(std::vector<ENERGYTYPE>& all_ener,
+               std::vector<std::vector<VALUETYPE> >& all_force,
+               std::vector<std::vector<VALUETYPE> >& all_virial,
+               const std::vector<VALUETYPE>& coord,
```
Comment (robinzyb) [inline source/api_cc/include/DeepPot.h:656]: I am consued that why the variables `coord` and `atype` are not the same as they are in the implementation `dcoord_`, `datype` @njzjz . I just follow the convention. Is it a typo or something else?

Comment (njzjz) [reply source/api_cc/include/DeepPot.h:656]: The convention exists in the initial version: https://github.com/deepmodeling/deepmd-kit/blame/e86cf20b4a9ad6ae629ec16c566ed3b086cb4b75/source/lib/src/NNPInter.cc
where dcoord is a vector and coord is a tensor. @wanghan-iapcm may answer.

**Review by njzjz — APPROVED** [review]
Looks good to me. I think you are expecting a new version, right?

**Comment by robinzyb** [comment]
> Looks good to me. I think you are expecting a new version, right?

Many thanks. Recently I will refactor the dpmd api in cp2k. Move the dpmd section from the forcefield section to a new section under force_eval.

**Pull request merged**